nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1647,13 +1620,26 @@ public boolean registerTrial() throws IOException, ConfigurationException, Packa
             return false;
         }
         log.info(String.format(
-                "Trial registered to %s for project %s\nDescription: %s\n"
+                "Trial registered to %s for project %s%n"
                         + "Please ensure you have validated your registration with the confirmation email before starting the server.",
-                registration.get(REGISTRATION_EMAIL), registration.get(REGISTRATION_PROJECT),
-                registration.get(REGISTRATION_DESCRIPTION)));
+                registration.get(TrialField.EMAIL.getId()), registration.get(TrialField.PROJECT.getId())));
         return true;
     }
 
+    protected void putIfValid(Map<String, String> map, String userInput, TrialField field)
+            throws ConfigurationException {
+        Predicate<String> isValid = field.getPredicate();
+        if (!isValid.test(userInput)) {
+            throw new ConfigurationException(field.getErrorMessage());
+        }
+        map.put(field.getId(), userInput);
+    }
+
+    protected void promptAndPut(Map<String, String> map, TrialField field) throws IOException, ConfigurationException {
```

Comment:
Let's discuss in [another thread](https://github.com/nuxeo/nuxeo/pull/873#discussion_r123192272).